diff -uNr libgcrypt-1.8.4/cipher/camellia-aarch64.S libgcrypt-1.8.4_patched/cipher/camellia-aarch64.S
--- libgcrypt-1.8.4/cipher/camellia-aarch64.S	2017-11-23 19:16:58.000000000 +0100
+++ libgcrypt-1.8.4_patched/cipher/camellia-aarch64.S	2019-03-22 18:06:37.635546976 +0100
@@ -208,7 +208,7 @@
 	 *	x3: keybitlen
 	 */
 
-	adr RTAB1,  _gcry_camellia_arm_tables;
+	ldr RTAB1, =_gcry_camellia_arm_tables;
 	mov RMASK, #(0xff<<4); /* byte mask */
 	add RTAB2, RTAB1, #(1 * 4);
 	add RTAB3, RTAB1, #(2 * 4);
@@ -251,7 +251,7 @@
 	 *	x3: keybitlen
 	 */
 
-	adr RTAB1,  _gcry_camellia_arm_tables;
+	ldr RTAB1, =_gcry_camellia_arm_tables;
 	mov RMASK, #(0xff<<4); /* byte mask */
 	add RTAB2, RTAB1, #(1 * 4);
 	add RTAB3, RTAB1, #(2 * 4);
