diff -uNr libssh2-1.10.0/src/CMakeLists.txt libssh2-1.10.0_patched/src/CMakeLists.txt
--- libssh2-1.10.0/src/CMakeLists.txt	2021-05-14 23:36:06.000000000 +0200
+++ libssh2-1.10.0_patched/src/CMakeLists.txt	2022-04-13 11:25:51.157773110 +0200
@@ -240,7 +240,9 @@
 add_feature_info(Compression ENABLE_ZLIB_COMPRESSION
   "using zlib for compression")
 if(ENABLE_ZLIB_COMPRESSION)
-  find_package(ZLIB REQUIRED)
+  find_package(net.zlib-zlib REQUIRED CONFIG)
+  GET_TARGET_PROPERTY(ZLIB_LIBRARIES net.zlib-zlib::zlibstatic LOCATION)
+  SET(ZLIB_INCLUDE_DIRS net.zlib-zlib_INCLUDE_DIR)
 
   target_include_directories(libssh2 PRIVATE ${ZLIB_INCLUDE_DIRS})
   list(APPEND LIBRARIES ${ZLIB_LIBRARIES})
